docs/development/mono-repository.md

```diff
@@ -0,0 +1,120 @@
+
+While there are a benefits, there are challenges to overcome.
+
+* Detecting what to build. Build times can be long due to overbuilding the entire repository
+* Long build times. Because of having to build the entire repository, this will cause build times to increase because tooling no longer is just building one artifact/package, but many different ones.
+* Not just JavaScript. A monorepo may not just be JavaScript, and can impact organization, tooling, and how to approach building different artifacts or packages in different languages.
+* Git repositories can grow larger in size.  Since we are building and maintaing different packages in a singular repository, we can exponentially grow the size of the git repository.
+* Commit messages can be harder to interpret due to multiple artifacts, packages, or teams.
+
+
+### Building
+
+When building npm packages (or any type of package) (the build) in the monorepo, we recommend:
+
+* [bazel][bazel]
+* Custom scripts
+
+Using appropriate build tools help overcome challenges state previously. With build tools, we are able to detect file changes and build the appropriate artifact/packages necessary without building the entire repository. This helps reduce long build times. Having a good build tool like [bazel][bazel] also allows for potential to build out other project artifact/packages in different languages.
+
+When committing messages, teams can utilize [conventionalcommits][conventionalcommits] to reduce confusion when working on a monorepo.
+
+### Publishing
+
+If you are working with a set of packages that need to be published, you can leverage:
+
+* [changesets][changesets]
+* [lerna][lerna] - Was previously used but because of [yarn workspaces][yarn-workspaces] and [npm workspaces][npm-workspaces], teams have no longer found a need for [lerna][lerna]
+
+#### Package Releasing
+
+We recommend following semantic versioning across all packages. If a change does not impact all packages, then bump the appropriate minor/patch version as necessary.
+
+For example, if a Package A needs security remediation, and Package B does not, simply increment patch version for Package A.
+
+If you are making a breaking change that is across all packages, apply major version to all packages.
+
+#### Service Releasing
+
+Service releasing is treated differently than publishing since services are released not for package consumption but as a service.
+
+Releases could be tagged differently per change or as an entire set (version of the repository).
+
+#### Advanced Tips / Tricks
+
+* TODO section blurb about hoisting and nohoist for strategies
```

Comment:
hoisting is not so relevant outside of lerna monorepos. rather than using the lerna-specific term `hoist`, it may be better to simply recommend that shared devDeps be installed in the root

Reply:
As far as I recall, hoisting is relevant still. You can inform yarn workspaces to hoist dependencies out or not hoist and maintain different versions of dependencies.